ui/src/TestAzureAccount.ts

```diff
@@ -0,0 +1,77 @@
+import { ArgumentError } from './errors';
+import { localize } from './localize';
+
+export class TestAzureAccount implements AzureAccount {
+    public status: AzureLoginStatus;
+    public onStatusChanged: Event<AzureLoginStatus>;
+    public waitForLogin: () => Promise<boolean>;
+    public sessions: AzureSession[];
+    public onSessionsChanged: Event<void>;
+    public subscriptions: AzureSubscription[];
+    public onSubscriptionsChanged: Event<void>;
+    public waitForSubscriptions: () => Promise<boolean>;
+    public filters: AzureResourceFilter[];
+    public onFiltersChanged: Event<void>;
+    public waitForFilters: () => Promise<boolean>;
+    private onStatusChangedEmitter: EventEmitter<AzureLoginStatus>;
+    private onFiltersChangedEmitter: EventEmitter<void>;
+
+    public constructor() {
+        this.subscriptions = [];
+        this.status = 'Initializing';
+        this.onStatusChangedEmitter = new EventEmitter<AzureLoginStatus>();
+        this.onStatusChanged = this.onStatusChangedEmitter.event;
+        this.onFiltersChangedEmitter = new EventEmitter<void>();
+        this.onFiltersChanged = this.onFiltersChangedEmitter.event;
+        this.filters = [];
+    }
+
+    public async getTestSubscription(): Promise<void> {
+        type servicePrincipalCredentials = ApplicationTokenCredentials & { environment: AzureEnvironment };
+        const clientId: string | undefined = process.env.SERVICE_PRINCIPAL_CLIENT_ID ;
+        const secret: string | undefined = process.env.SERVICE_PRINCIPAL_SECRET;
+        const domain: string | undefined = process.env.SERVICE_PRINCIPAL_DOMAIN;
+        if (!clientId || !secret || !domain) {
+            throw new Error(localize('travisOnly', 'TestAzureAccount cannot be used without the following environment variables: SERVICE_PRINCIPAL_CLIENT_ID, SERVICE_PRINCIPAL_SECRET, SERVICE_PRINCIPAL_DOMAIN'));
+        }
+        this.status = 'LoggingIn';
+        const credentials: servicePrincipalCredentials = <servicePrincipalCredentials>(await loginWithServicePrincipalSecret(clientId, secret, domain));
+        const subscriptionClient: SubscriptionClient = new SubscriptionClient(credentials);
+        const subscriptions: SubscriptionListResult = await subscriptionClient.subscriptions.list();
+        // returns an array withy subscriptionId, displayName
+        const tenants: TenantListResult = await subscriptionClient.tenants.list();
+
+        if (tenants[0].id) {
+            const tenantId: string = <string>tenants[0].id;
+            const session: AzureSession = {
+                environment: credentials.environment,
+                userId: '',
+                tenantId: tenantId,
+                credentials: credentials
+            };
+
+            // tslint:disable-next-line:strict-boolean-expressions
+            if (subscriptions && subscriptions[0].id && subscriptions[0].displayName && subscriptions[0].subscriptionId) {
```

Comment:
nit: Why do you need to disable `strict-boolean-expressions` here? I'm guessing you can remove `subscriptions` from the if clause because it's probably always defined